src/odemis/driver/technolution.py

```diff
@@ -0,0 +1,612 @@
+from __future__ import division
+import json
+import queue
+import re
+from datetime import datetime
+import logging
+import time
+import threading
+import requests
+import numpy
+from odemis import model
+import weakref
+
+from odemis import model
+from src.openapi_server.models.mega_field_meta_data import MegaFieldMetaData
+from src.openapi_server.models.cell_parameters import CellParameters as CellAcqParameters
+from src.openapi_server.models.field_meta_data import FieldMetaData
+from src.openapi_server.models.error_status import ErrorStatus
+
+# TODO K.K. remove set logging level
+logging.getLogger().setLevel(logging.DEBUG)
+
+# Variable to differentiate "get" and "post" requests to the ASM server
+_METHOD_GET = 1
+_METHOD_POST = 2
+
+DATA_CONTENT_TO_ASM = {"empty": None, "thumbnail": True, "full": False}
+
+
+class AcquisitionServer(model.HwComponent):
+
+    def __init__(self, name, role, server_url, children={}, deamon=None, **kwargs):
+        super(AcquisitionServer, self).__init__(name, role, **kwargs)
+
+        self.server_url = server_url
+        # TODO K.K. Set external storage and connection
+
+        # Stop any acquisition if already one was in progress
+        self.ASMAPICall(self.server_url + "/scan/finish_mega_field", _METHOD_POST, 204)
+
+        # TODO K.K. remove *kwargs and include setup of entire component in try (each sperated) first check if childeren
+        # holds value of component
+        try:
+            kwargs = children["MirrorDescanner"]
+        except Exception:
+            raise ValueError("Required child MirrorDescanner not provided")
+        self._mirror_descanner = MirrorDescanner("MirrorDescanner", role=None, parent=self)
+        self.children.value.add(self._mirror_descanner)
+
+        try:
+            kwargs = children["EBeamScanner"]
+        except Exception:
+            raise ValueError("Required child EBeamScanner not provided")
+        self._ebeam_scanner = EBeamScanner("EBeamScanner", role=None, parent=self)
+        self.children.value.add(self._ebeam_scanner)
+
+        try:
+            kwargs = children["MPPC"]
+        except Exception:
+            raise ValueError("Required child MPPC not provided")
+        self._mppc = MPPC("MPPC", role=None, parent=self)
+        self.children.value.add(self._mppc)
+
+    @classmethod
+    def ASMAPICall(cls, url, method, expected_status, data=None, raw_response=False, timeout=600):
+        """
```

Comment:
I think it'd be easier to use and read if this was actually two separate functions, one for each method. Ex: "ASMGet" and "ASMPost". 
Also, don't make it a classmethod, but a simple method. In this case, you can append `self._server_url` by itself, which will again simplify the calls.
All the places which use the classmethod are in the test cases, and that's not a good reason because the test case shouldn't rely on the internal of the object. => Either first instantiate the class, or call directly the hardware with `request.*`. For automatically stopping the acquisition at init, put it in the class init, if that's useful, and drop it completely if it's not.